### PR TITLE
Simplify `getSupportInfo`

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/main/support.js [hunk 1]
```diff
@@ -133,19 +107,16 @@ function getSupportInfo(version, opts) {
     if (!object.deprecated || opts.showDeprecated) {
       return object;
     }
-    const newObject = Object.assign({}, object);
-    delete newObject.deprecated;
-    delete newObject.redirect;
+    // eslint-disable-next-line no-unused-vars
+    const { deprecated, redirect, ...newObject } = Object.assign({}, object);
```
Comment (thorn0) [inline src/main/support.js]: Let's enable https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings , shall we?

Comment (fisker) [reply src/main/support.js]: I thought about it, but I thought this case 

```js
const {a, ...b} = c
```

if a and b both not used, it  will not report an error.

But I'm not sure now.. not near my laptop.

Comment (thorn0) [reply src/main/support.js]: BTW, `Object.assign` is not needed here.

Comment (fisker) [reply src/main/support.js]: Oops, missed this one.

Comment (fisker) [reply src/main/support.js]: Fixed.

Also enabled ignoreRestSiblings,

```js
const {a, ...b} = c
```

If a&b not used, report b is not used.
If a used b not used, report b is not used.
If b used a not used, no error.

File: src/main/support.js [hunk 2]
```diff
@@ -25,43 +23,33 @@ function getSupportInfo(version, opts) {
...
-      ),
-      coreOptions
-    ),
+    Object.assign({}, ...plugins.map(({ options }) => options), coreOptions),
     "name"
   )
-    .sort((a, b) => (a.name === b.name ? 0 : a.name < b.name ? -1 : 1))
     .filter(option => filterSince(option) && filterDeprecated(option))
-    .map(mapDeprecated)
-    .map(mapInternal)
```
Comment (thorn0) [inline src/main/support.js]: This looked good. Why change?

Comment (fisker) [reply src/main/support.js]: Old one `map` 3 times, generate two extra new array, should merge into one `map`, for better performance.

Comment (thorn0) [reply src/main/support.js]: This place is not performance-critical, so please revert.

File: src/main/support.js [hunk 3]
```diff
@@ -25,43 +23,33 @@ function getSupportInfo(version, opts) {
...
+    .sort((a, b) => (a.name === b.name ? 0 : a.name < b.name ? -1 : 1))
```
Comment (fisker) [inline src/main/support.js]: I changed `filter` and `sort` order, so array for `sort` could be smaller.

**Pull request merged**